src/libsync/filesystem.cpp

```diff
@@ -394,4 +402,58 @@ QString FileSystem::fileSystemForPath(const QString & path)
 }
 #endif
 
+QByteArray FileSystem::calcMd5( const QString& filename )
+{
+    QByteArray arr;
+
+    QCryptographicHash crypto( QCryptographicHash::Md5 );
+
+    QFile file(filename);
+    if (file.open(QIODevice::ReadOnly)) {
+        QByteArray data;
+        while (!file.atEnd()) {
+            data = file.read(1024*1024*10);
```

Comment:
Is is much more efficiant to call the  read overload that takes a char* and put data in a buffer.  Because this re-allocates a new buffer all the time.